src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13587,6 +13585,29 @@ public void LambdaWithParameterDefaultValueAttribute_NoOptional()
                 Diagnostic(ErrorCode.ERR_NoCorrespondingArgument, "inferred").WithArguments("arg", "System.Func<int, int>").WithLocation(12, 1));
         }
 
+        [Fact]
+        public void LambdaWithDefaultParameterValueAttribute_SynthesizedDelegateTypeMatch()
+        {
+            var source = """
+                using System.Runtime.InteropServices;
+                var lam1 = (int a = 1) => a;
+                var lam2 = ([Optional] int b) => b;
+                var lam3 = ([DefaultParameterValue(1)] int c) => c;
+                var lam4 = ([Optional, DefaultParameterValue(1)] int d) => d;
+                Report(lam1);
+                Report(lam2);
+                Report(lam3);
+                Report(lam4);
+                static void Report(object obj) => System.Console.WriteLine(obj.GetType());
+                """;
+            CompileAndVerify(source, expectedOutput: """
+                <>f__AnonymousDelegate0
+                System.Func`2[System.Int32,System.Int32]
+                System.Func`2[System.Int32,System.Int32]
+                <>f__AnonymousDelegate0
```

Comment:
Do we already have a test showing that a different default value prevents sharing of synthesized delegate types? Consider add `var lam5 = (int a = 42) => a;` to this test.